packages/react-storage/src/components/StorageBrowser/context/actions/listLocationItemsAction.ts

```diff
@@ -21,15 +21,50 @@ export interface ListLocationItemsActionOutput
 
 type ListOutputItem = ListPaginateWithPathOutput['items'][number];
 
-const parseResultItems = (items: ListOutputItem[]): LocationItem[] =>
+const parseResultItems = (
+  items: ListOutputItem[],
+  path: string
+): LocationItem[] =>
   items.map(({ path: key, lastModified, size }) => {
+    const keyWithoutPath = key.slice(path.length);
     if (size === 0 && key.endsWith('/')) {
-      return { key, type: 'FOLDER' };
+      return { key: keyWithoutPath, type: 'FOLDER' };
     }
 
-    return { key, lastModified: lastModified!, size: size!, type: 'FILE' };
+    return {
+      key: keyWithoutPath,
+      lastModified: lastModified!,
+      size: size!,
+      type: 'FILE',
+    };
   });
```

Comment:
Minor readability cleanup:
```suggestion
const parseResultItems = (
  items: ListOutputItem[],
  path: string
): LocationItem[] =>
  items.map(({ path: _path, lastModified, size }) => {
    const key = _path.slice(path.length);
    if (size === 0 && key.endsWith('/')) {
      return { key, type: 'FOLDER' };
    }

    return {
      key,
      lastModified: lastModified!,
      size: size!,
      type: 'FILE',
    };
  });
```

Reply:
fixed